master/internal/command/command.go

```diff
@@ -230,11 +232,11 @@ func (c *command) Receive(ctx *actor.Context) error {
 				})
 				names = append(names, string(c.taskID))
 			}
-			if c.assignedPort == 0 && len(names) > 0 {
-				ctx.Log().Error("Expected to not proxy any ports but proxied one anyway")
+			if c.assignedPort == nil && len(names) > 0 {
+				ctx.Log().Error("expected to not proxy any ports but proxied one anyway")
 			} else if len(names) != 1 {
 				ctx.Log().Errorf(
```

Comment:
this is ancillary to this pr, but it seems both of these cases should crash the actor to me.

Reply:
I think it's better to fail a bit more gracefully.  For example, if we had been that drastic before, then the customer who was hit this originally would have had absolutely no recourse and would not have been able to continue using their notebook by bypassing the master proxy.

It's not like the user isn't going to know that the notebook failed; that will still be obvious.

Though I do agree it's a bad error condition that I would definitely want to see when I asked a customer for the master logs.

Reply:
yeah.. degrading in unanticipated ways just always seems dangerous (scary) to me, but I can see that case at least.